Informatics I/010 Modularization and Git/Task 1 International Banking/Perytron Solution/currency_converter.py

```diff
@@ -0,0 +1,29 @@
+from typing import Union
+from exchange_rates import EXCHANGE_RATES
+
+def convert(amount: Union[int, float], from_currency: str, to_currency: str) -> float:
+
+    # Check that correct type is provided
+    if False in [isinstance(from_currency, str), isinstance(to_currency, str), (isinstance(amount, float) or isinstance(amount, int))]: raise Warning("You had one (...) job. Provide the right type you (...) (...)!")
+
+    # Check that the amount is greater than 0
+    if amount <= 0: raise Warning("Sigh... You clearly won't pass Mikro.")
+
+    # If exchange rate exists without calculating it
+    try: EXCHANGE_RATES[from_currency][to_currency]
+    except: pass
+    else: return amount * EXCHANGE_RATES[from_currency][to_currency]
```

Comment:
wokring solution, possible enhancement: put return into try, will work the same but with one line less.